src/firmware/guest/mod.rs

```diff
@@ -86,23 +86,62 @@ impl Firmware {
         &mut self,
         message_version: Option<u8>,
         report_request: SnpReportReq,
-    ) -> Result<(AttestationReport, CertTable), Indeterminate<Error>> {
+    ) -> Result<(AttestationReport, Vec<CertTableEntry>), Indeterminate<Error>> {
+        // Due to the complex buffer allocation, we will take the SnpReportReq
+        // provided by the caller, and create an extended report request object
+        // for them.
         let ext_report_request: SnpExtReportReq = SnpExtReportReq::new(report_request);
```

Comment:
1. This creates an SnpExtReportReq in which both certs_address and certs_len are zero. This will cause the sev-guest ioctl to not fill in certificates information at all (see [sev-guest.c](https://github.com/torvalds/linux/blob/7287904c8771b77b9504f53623bb477065c19a58/drivers/virt/coco/sev-guest/sev-guest.c#L504-L506) in the kernel).

   In AMD's sev-guest repo in get-report.c they set `certs_address = (u64)-1` and `certs_len = 0` to signal to the kernel module that the caller wants to know the needed length.

   Maybe modify `SnpExtReportReq::new` to put `u64::MAX` in `certs_address` ?

2. Should this variable be `mut` because the ioctl might modify its content?

Reply:
> 1. This creates an SnpExtReportReq in which both certs_address and certs_len are zero. This will cause the sev-guest ioctl to not fill in certificates information at all (see [sev-guest.c](https://github.com/torvalds/linux/blob/7287904c8771b77b9504f53623bb477065c19a58/drivers/virt/coco/sev-guest/sev-guest.c#L504-L506) in the kernel).
>    In AMD's sev-guest repo in get-report.c they set `certs_address = (u64)-1` and `certs_len = 0` to signal to the kernel module that the caller wants to know the needed length.
>    Maybe modify `SnpExtReportReq::new` to put `u64::MAX` in `certs_address` ?

Yes, I think this would work well.

> 2. Should this variable be `mut` because the ioctl might modify its content?

I believe it should.

Reply:
I went with this approach in the end. Thank you for reminding me about it.